test/test_sourceform.py

```diff
@@ -512,6 +513,99 @@ class (nuz), intent(in) :: this_
 
         assert call.name == expected_name
 
+
+def test_submodule_private_var_call(parse_fortran_file):
+    data = f"""\
+    module foo
+      type :: nuz
+        integer :: var
+      end type nuz
+
+      type(nuz), dimension(2), private :: pri_var
+
+      interface
+        module subroutine bar()
+        end subroutine bar
+      end interface
+    end module foo
+
+    submodule(foo) baz
+    contains
+      module procedure bar
+        integer :: val
+        val = pri_var(1)%var
+      end procedure bar
+    end submodule baz
+    """
+    expected = []
+
+    fortran_file = parse_fortran_file(data)
+    fp = FakeProject()
+    modules = {
+        module.name: module
+        for module in chain(fortran_file.modules, fortran_file.submodules)
+    }
+    for module in modules.values():
+        find_used_modules(module, modules.values(), [], [])
+
+    # correlation order is important
+    modules["foo"].correlate(fp)
+    modules["baz"].correlate(fp)
+
+    main_subroutines = {sub.name: sub for sub in modules["baz"].modprocedures}
+    calls = main_subroutines["bar"].calls
+
+    assert len(calls) == len(expected)
+
+    calls_sorted = sorted(calls, key=lambda x: getattr(x, "name", x))
+    expected_sorted = sorted(expected)
+    for call, expected_name in zip(calls_sorted, expected_sorted):
+        assert isinstance(call, FortranBase)
+
+        assert call.name == expected_name
```

Comment:
I guess this test is like this as it's essentially a copy of another test. I think this would be clearer with just:

```suggestion
    assert calls == []
```

unless we refactor all these similar tests into a parameterised test? That might be more effort than it's worth?